x-pack/platform/plugins/shared/cases/server/saved_object_types/import_export/import.ts

```diff
@@ -0,0 +1,28 @@
+/*
+ * Copyright Elasticsearch B.V. and/or licensed to Elasticsearch B.V. under one
+ * or more contributor license agreements. Licensed under the Elastic License
+ * 2.0; you may not use this file except in compliance with the Elastic License
+ * 2.0.
+ */
+
+import type { Logger, SavedObject, SavedObjectsImportHookResult } from '@kbn/core/server';
+import type { CasePersistedAttributes } from '../../common/types/case';
+import { createCaseError } from '../../common/error';
+
+export function handleImport({
+  objects,
+  logger,
+}: {
+  objects: Array<SavedObject<CasePersistedAttributes>>;
+  logger: Logger;
+}): SavedObjectsImportHookResult {
+  const hasObjectsWithIncrementalId = objects.some((obj) => !!obj.attributes.incremental_id);
+  if (hasObjectsWithIncrementalId) {
+    throw createCaseError({
+      message: 'Remove `incremental_id` from cases before importing',
+      logger,
+    });
+  } else {
+    return {};
+  }
```

Comment:
It's an edge case, but `incremental_id: 0` would be considered valid. Could we use `obj.attributes.incremental_id != null` instead of `!!`?

About the error message, consider just notifying of the problem, rather than asking for a specific action:

```ts
message: 'The incremental_id field is not supported on importing',
```

Reply:
Nice catch, fixed here https://github.com/elastic/kibana/pull/219757/commits/0ed8e25b29542715e7f969c5600868c8738cd441